File: ollama/codigo-final/fasta/codegemma-7b/fewshot/temp_0/fasta.java
import java.util.Random;

public class FastaGenerator {
    public static void main(String[] args) {
        int N = Integer.parseInt(args[0]);

        // LCG constants
        long IM = 139968;
        long IA = 3877;
        long IC = 29573;
        long Seed = 42;

        // ALU sequence
        String alu = "GGCCGGGCGCGGTGGCTCACGCCTGTAATCCCAGCACTTTGGGAGGCCGAGGCGGGCGGA" +
                "TCACCTGAGGTCAGGAGTTCGAGACCAGCCTGGCCAACATGGTGAAACCCCGTCTCTACT" +
                "AAAAATACAAAAATTAGCCGGGCGTGGTGGCGCGCGCCTGTAATCCCAGCTACTCGGGAG" +
                "GCTGAGGCAGGAGAATCGCTTGAACCCGGGAGGCGGAGGTTGCAGTGAGCCGAGATCGCG" +
                "CCACTGCACTCCAGCCTGGGCGACAGAGCGAGACTCCGTCTCAAAAA";

        // IUB ambiguity codes and probabilities
        double[] iub = {0.27, 0.12, 0.12, 0.27, 0.02, 0.02, 0.02, 0.02, 0.02};

        // Homo sapiens probabilities
        double[] sapiens = {0.3029549426680, 0.1979883004921, 0.1975473066391, 0.3015094502008};

        // Random number generator
        Random random = new Random((int) Seed);

        // Generate sequences
        System.out.println(">ONE Homo sapiens alu");
        for (int i = 0; i < N * 2; i++) {
            System.out.print(alu.charAt(random.nextInt(alu.length())));
            if (i % 60 == 59) {
                System.out.println();
            }
        }

        System.out.println(">TWO IUB ambiguity codes");
        for (int i = 0; i < N * 3; i++) {
            double r = random.nextDouble();
            int index = 0;
            for (double p : iub) {
                if (r < p) {
                    System.out.print(indexToNucleotide(index));
                    break;
                }
                r -= p;
                index++;
            }
            if (i % 60 == 59) {
                System.out.println();
            }
        }

        System.out.println(">THREE Homo sapiens frequency");
        for (int i = 0; i < N * 5; i++) {
            double r = random.nextDouble();
            int index = 0;
            for (double p : sapiens) {
                if (r < p) {
                    System.out.print(indexToNucleotide(index));
                    break;
                }
                r -= p;
                index++;
            }
            if (i % 60 == 59) {
                System.out.println();
            }
        }
    }

    // Convert integer index to nucleotide
    private static char indexToNucleotide(int index) {
        switch (index) {
            case 0: return 'a';
            case 1: return 'c';
            case 2: return 'g';
            case 3: return 't';
            case 4: return 'B';
            case 5: return 'D';
            case 6: return 'H';
            case 7: return 'K';
            case 8: return 'M';
            case 9: return 'N';
            case 10: return 'R';
            case 11: return 'S';
            case 12: return 'V';
            case 13: return 'W';
            case 14: return 'Y';
            default: throw new IllegalStateException("Invalid index");
        }
    }
}

File: ollama/codigo-final/fasta/codegemma-7b/fewshot/temp_0/fasta.cpp
#include <iostream>
#include <string>
#include <vector>

struct RandomGenerator {
    int seed;
    double random(double max) {
        seed = ((seed * IA + IC) % IM);
        return max * seed / IM;
    }
};

constexpr int IM = 139968, IA = 3877, IC = 29573;

std::string generateRepeatFasta(const std::string& pattern, int n) {
    std::string result;
    for (int i = 0; i < n; i++) {
        result += pattern;
    }
    return result;
}

std::string generateRandomFasta(const std::vector<std::pair<char, double>>& alphabet,
                               int length) {
    RandomGenerator rng{42};
    std::string result;
    for (int i = 0; i < length; i++) {
        double r = rng.random(1.0);
        int index = 0;
        for (const auto& [_, p] : alphabet) {
            if (r < p) {
                result += alphabet[index].first;
                break;
            }
            r -= p;
            index++;
        }
    }
    return result;
}

int main(int argc, char *argv[]) {
    if (argc != 2) {
        std::cerr << "Usage: " << argv[0] << " N" << std::endl;
        return 1;
    }
    int N = std::stoi(argv[1]);

    const std::string ALU_SEQUENCE = "GGCCGGGCGCGGTGGCTCACGCCTGTAATCCCAGCACTTTGGGAGGCCGAGGCGGGCGGA" +
                                   "TCACCTGAGGTCAGGAGTTCGAGACCAGCCTGGCCAACATGGTGAAACCCCGTCTCTACT" +
                                   "AAAAATACAAAAATTAGCCGGGCGTGGTGGCGCGCGCCTGTAATCCCAGCTACTCGGGAG" +
                                   "GCTGAGGCAGGAGAATCGCTTGAACCCGGGAGGCGGAGGTTGCAGTGAGCCGAGATCGCG" +
                                   "CCACTGCACTCCAGCCTGGGCGACAGAGCGAGACTCCGTCTCAAAAA";

    const std::vector<std::pair<char, double>> IUB_ALPHABET = {
        {'a', 0.27}, {'c', 0.12}, {'g', 0.12}, {'t', 0.27},
        {'B', 0.02}, {'D', 0.02}, {'H', 0.02}, {'K', 0.02},
        {'M', 0.02}, {'N', 0.02}, {'R', 0.02}, {'S', 0.02},
        {'V', 0.02}, {'W', 0.02}, {'Y', 0.02}
    };

    const std::vector<std::pair<char, double>> HSAP_ALPHABET = {
        {'a', 0.3029549426680}, {'c', 0.1979883004921},
        {'g', 0.1975473066391}, {'t', 0.3015094502008}
    };

    auto repeat_alu = generateRepeatFasta(ALU_SEQUENCE, N * 2);
    auto random_iub   = generateRandomFasta(IUB_ALPHABET, N * 3);
    auto random_hsap  = generateRandomFasta(HSAP_ALPHABET, N * 5);

    std::cout << ">ONE Homo sapiens alu\n" << repeat_alu << "\n";
    std::cout << ">TWO IUB ambiguity codes\n" << random_iub << "\n";
    std::cout << ">THREE Homo sapiens frequency\n" << random_hsap;

    return 0;
}
